examples/lang2/erc721/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(proc_macro_hygiene)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang2 as ink;
+use ink_core::storage;
+
+#[ink::contract(version = "0.1.0")]
+mod erc721 {
+    #[ink(storage)]
+    struct Erc721 {
+        /// Mapping from token to owner.
+        token_owner: storage::HashMap<Hash, AccountId>,
+        /// Mapping from token to approvals users.
+        token_approvals: storage::HashMap<Hash, AccountId>,
+        /// Mapping from owner to number of owned token.
+        owned_tokens_count: storage::HashMap<AccountId, u128>,
+        /// Mapping from owner to operator approvals
+        operator_approvals: storage::HashMap<(AccountId, AccountId), bool>
+    }
```

Comment:
Due to being compiled for `wasm32` which is a 32-bit architecture we can simplify certain parts of this. For example you could use a `struct TokenId(u32)` instead of `Hash` for the token ID which is just 32 bit instead of 256 bit large structure. That should help in reducing gas costs.
Also `owned_tokens_count` could therefore be a mapping from `AccountId -> u32` instead of `u128`.